### PR TITLE
Write int96 to parquet

### PR DESCRIPTION
**Pull request by razajafri**
<s>This is an early WIP. I think it's pretty much complete but still needs to be tested after the chunked writer accepts INT96 option</s>

This is now ready for review.

### CODE REVIEW

**Comment by sameerz** [comment]
Related cudf PR https://github.com/rapidsai/cudf/pull/6625

**Comment by revans2** [comment]
This also depends on https://github.com/rapidsai/cudf/pull/6679

**Comment by razajafri** [comment]
build'

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
I apologize for the inconvenience but I am using the CI to run the tests as my host is not running the tests

**Comment by razajafri** [comment]
build

**Review by jlowe — CHANGES_REQUESTED** [review]

File: integration_tests/src/main/python/parquet_write_test.py [hunk 1]
```diff
@@ -186,22 +184,6 @@ def test_parquet_write_legacy_fallback(spark_tmp_path, ts_write, ts_rebase, spar
             'DataWritingCommandExec',
             conf=all_confs)
 
-@allow_non_gpu('DataWritingCommandExec')
-@pytest.mark.parametrize('ts_write', ['INT96'])
```
[the remark this answers is not in the document]
Comment (razajafri) [reply integration_tests/src/main/python/parquet_write_test.py:190]: Thanks for the comment @jlowe. My changes to the test were clobbered by a merge, looks like I was a careless carrying changes over to this file. I have re-ported the changes, PTAL again. I also confused the `int96RebaseModeInWrite` and `datetimeRebaseModeInWrite`. The former is needed for 3.1.0 tests and I have added it now. We support `CORRECTED` and `EXCEPTION` only. One thing that I hadn't included in my original change was to test that the EXCEPTION mode actually fails. I have also added that. 

Please let me know if you want me to add anything more.

**Comment by razajafri** [comment]
build

File: integration_tests/src/main/python/parquet_write_test.py [hunk 2]
```diff
@@ -124,32 +130,31 @@ def write_parquet_sql_from(spark, df, data_path, write_to_table):
...
 
 # TODO - https://github.com/NVIDIA/spark-rapids/issues/1130 to handle TIMESTAMP_MILLIS
-parquet_ts_write_options = ['TIMESTAMP_MICROS']
-
-@pytest.mark.parametrize('ts_write', parquet_ts_write_options)
+@pytest.mark.parametrize('ts_write', ['INT96', 'TIMESTAMP_MICROS'])
 @pytest.mark.parametrize('ts_rebase', ['EXCEPTION'])
 def test_ts_write_fails_datetime_exception(spark_tmp_path, ts_write, ts_rebase, spark_tmp_table_factory):
-    gen = TimestampGen(start=datetime(1590, 1, 1, tzinfo=timezone.utc))
+    gen = TimestampGen(start=datetime(1, 1, 1, tzinfo=timezone.utc), end=datetime(1582, 1, 1, tzinfo=timezone.utc))
```
Comment (revans2) [inline integration_tests/src/main/python/parquet_write_test.py:157]: Why did these date ranges change?

Comment (razajafri) [reply integration_tests/src/main/python/parquet_write_test.py:157]: Correct me if I am wrong, but to be sure an exception is thrown, the dates should be before 1582 and timestamps before 1900? This is to make sure that we are generating dates that will throw an exception.

**Review by revans2 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Pull request merged**